src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -124,6 +126,44 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
         return tagSet;
     }
 
+    /**
+     * Parses a {@code String status} into a {@code DeliveryStatus}.
+     *
+     * @param status the status to be parsed.
+     * @return the parsed DeliveryStatus.
+     * @throws ParseException if the given {@code status} is invalid.
+     */
+    public static DeliveryStatus parseDeliveryStatus(String status) throws ParseException {
+        requireNonNull(status);
+        String trimmedStatus = status.trim().toUpperCase();
+
+        if (trimmedStatus.equals("ALL")) {
+            return null;
```

Comment:
This might result in nullPointer for classes dependent on status

Reply:
What would be a better approach though to indicate "All"?

Reply:
Possibly another enum, but I think can be left for future increment